pkg/ko/build/gobuild.go

```diff
@@ -132,7 +132,7 @@ func tarBinary(binary string) (*bytes.Buffer, error) {
 		return nil, err
 	}
 	header := &tar.Header{
-		Name:     appPath,
+		Name:     name,
```

Comment:
Any issues with just creating the file here without explicitly creating/setting permissions on the parent directory? I don't see any in my testing.

Reply:
I think containerd gets mad if you don't, is that right? @dlorenc

Reply:
I don't think it gets mad, it just creates missing directories with a different set of permissions than docker. If you dont care about the permissions of these parent directories, it should just work. If you care, it's best to create it explicitly.

Reply:
Containerd was creating the directory with 0700 permissions, which seems like it would be an issue when running as anything but root.

I explicitly create the parent directory now with 0555 permissions.